streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -220,12 +215,18 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
             } else {
                 try {
                     task.suspend();
-                    final Map<TopicPartition, OffsetAndMetadata> committableOffsets = task.prepareCommit();
-
-                    tasksToClose.add(task);
-                    if (!committableOffsets.isEmpty()) {
-                        consumedOffsetsAndMetadataPerTask.put(task.id(), committableOffsets);
+                    if (task.commitNeeded()) {
+                        if (task.isActive()) {
+                            log.error("Active task {} was revoked and should have already been committed", task.id());
+                            throw new IllegalStateException("Revoked active task was not committed during handleRevocation");
+                        } else {
+                            task.prepareCommit();
+                            task.postCommit();
+                        }
                     }
+                    completeTaskCloseClean(task);
```

Comment:
When closing-clean a standby task, we would checkpoint the file and close the state store which would also flush it as well, so I think we do not need to call

```
task.prepareCommit();
task.postCommit();
```

which is just to flush the stores and write checkpoint files, right?

Reply:
Actually we don't checkpoint during `closeClean` anymore. You always have to commit (if `clean` && `commitNeeded`) before closing

Reply:
Is there a different PR removing that? I still see

```
if (clean) {
                // since there's no written offsets we can checkpoint with empty map,
                // and the state current offset would be used to checkpoint
                stateMgr.checkpoint(Collections.emptyMap());
                offsetSnapshotSinceLastCommit = new HashMap<>(stateMgr.changelogOffsets());
            }
```

in Standby.close() in trunk.

Reply:
I suspect your trunk is out of date 🙂 

(that code & comment is now in `postCommit`)

Reply:
You're right :)